packages/headless/src/features/generated-answer/generated-answer-actions.ts

```diff
@@ -147,7 +147,7 @@ export const streamAnswer = createAsyncThunk<
       write: (data: GeneratedAnswerStreamEventData) => {
         if (
           request.streamId ===
-          state.search.extendedResults.generativeQuestionAnsweringId
+          config.getState().search.extendedResults.generativeQuestionAnsweringId
```

Comment:
I guess a simple naming changes could make this more obvious to a future reader, to not fall into the same trap (what the different closure are in this `streamAnswer` asynchronous function)

Suggestion: an inline utility function

```typescript
if(currentStreamMatchesOriginalStream(originalState, currentState) {
     // ...
}
```

Reply:
That's a good point, especially since this is repeated 4 times, I would make a function with a clear name indeed.